org.civicrm.reqangular/src/common/models/instances/instance.js

```diff
@@ -1,117 +1,130 @@
+       */
+      fromAPI: function (attributes) {
+        return _.transform(
+          attributes,
+          this.fromAPIFilter.bind(attributes),
+          Object.create(null)
+        );
+      },
 
-            /**
-             * Function that filters data coming from the API, being called in
-             * as the iteratee of a _.transform() call
-             *
-             * Can be overriden by children api for custom filtering
-             *
-             * @param {object} result - The accumulator object
-             * @param {string} key - The property name
-             */
-            fromAPIFilter: function (result, __, key) {
-                result[key] = this[key];
-            },
+      /**
+       * Function that filters data coming from the API, being called in
+       * as the iteratee of a _.transform() call
+       *
+       * Can be overriden by children api for custom filtering
+       *
+       * @param {object} result - The accumulator object
+       * @param {string} key - The property name
+       */
+      fromAPIFilter: function (result, __, key) {
+        result[key] = this[key];
+      },
 
-            /**
-             * Creates a new instance, optionally with its data normalized
-             *
-             * @param {object} attributes - The instance data
-             * @param {boolean} fromAPI - If the data comes from the API and needs to be normalized
-             * @return {object}
-             */
-            init: function (attributes, fromAPI) {
-                attributes = _.assign(this.defaultCustomData(), attributes);
+      /**
+       * Creates a new instance, optionally with its data normalized.
+       * Also, it will allow children to add/remove/update current attributes of
+       * the instance using customInit method
+       *
+       * @param {object} attributes - The instance data
+       * @param {boolean} fromAPI - If the data comes from the API and needs to be normalized
+       * @return {object}
+       */
+      init: function (attributes, fromAPI) {
+        if (typeof fromAPI !== 'undefined' && fromAPI) {
+          attributes = this.fromAPI(attributes);
+        }
+        attributes = _.assign(this.defaultCustomData(), attributes);
+        attributes = this.customInit(attributes);
 
-                if (typeof fromAPI !== 'undefined' && fromAPI) {
-                    attributes = this.fromAPI(attributes);
-                }
+        return _.assign(Object.create(this), attributes);
+      },
 
-                return _.assign(Object.create(this), attributes);
-            },
+      /**
+       * Method that a children instance can redefine to perform custom initialization logic
+       * The children can add/remove/update the attributes of this instance and return the same
+       *
+       * @param {object} attributes - The instance data
+       * @return {object} updated instance attributes object
+       */
+      customInit: function (attributes) {
```

Comment:
We agreed that `customInit` is not really a good name for what this method is for